packages/upload-api/src/space-allocate.js

```diff
@@ -1,19 +1,33 @@
 import * as API from './types.js'
 import * as Server from '@ucanto/server'
+import * as Ucanto from '@ucanto/interface'
 import * as Space from '@web3-storage/capabilities/space'
+import { ensureRateLimitAbove } from './utils/rate-limits.js'
 
 /**
  *
- * @param {{capability: {with: API.SpaceDID, nb:{size:number}}}} input
+ * @param {{capability: {with: API.SpaceDID}}} input
  * @param {API.SpaceServiceContext} context
- * @returns {Promise<API.Result<{size:number}, API.AllocationError>>}
+ * @returns {Promise<API.Result<Ucanto.Unit, API.AllocationError>>}
  */
 export const allocate = async ({ capability }, context) => {
-  const { with: space, nb } = capability
-  const { size } = nb
+  const { with: space } = capability
+  const rateLimitResult = await ensureRateLimitAbove(
+    context.rateLimitsStorage,
+    [space],
+    0
+  )
+  if (rateLimitResult.error) {
+    return {
+      error: {
+        name: 'InsufficientStorage',
```

Comment:
`InsufficientStorage` doens't feel like the right name here. It's more of an `AccountBlocked` kind of error.